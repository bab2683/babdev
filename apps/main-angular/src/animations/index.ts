export * from './header.animation';
export * from './menu.animation';
export * from './router.animation';
