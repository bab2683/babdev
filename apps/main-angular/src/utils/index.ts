export * from './date.utils';
export * from './page.utils';
export * from './string.utils';
