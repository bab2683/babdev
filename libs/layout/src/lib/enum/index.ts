export * from './background.enum';
export * from './clickable.enum';
export * from './tooltip.enum';
