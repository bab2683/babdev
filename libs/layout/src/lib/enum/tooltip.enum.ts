export enum TooltipStatus {
  CLOSED = 'closed',
  OPEN = 'open'
}
