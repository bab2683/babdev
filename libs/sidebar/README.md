# sidebar

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test sidebar` to execute the unit tests.

## Customization

In order to change the appearance of the library, some css variables can be changed:
* **--babdev-sidebar-animation**: the animation style of the sidebar (default: `300ms ease`).
* **--babdev-sidebar-background**: the background color of the sidebar (default: `#e2e2e2`).
* **--babdev-sidebar-overlay-color**: the color of the overlay, if present (default: `rgba(0, 0, 0, 0.4)`)
* **--babdev-sidebar-padding**: the padding of the sidebar (default: `10px`).
* **--babdev-sidebar-width**: the width of the sidebar (default: `auto`).
