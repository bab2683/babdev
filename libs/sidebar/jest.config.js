module.exports = {
  name: 'sidebar',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/libs/sidebar',
  snapshotSerializers: [
    'jest-preset-angular/AngularSnapshotSerializer.js',
    'jest-preset-angular/HTMLCommentSerializer.js'
  ]
};
