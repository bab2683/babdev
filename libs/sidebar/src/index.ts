export * from './lib/sidebar.component';
export * from './lib/sidebar.enum';
export * from './lib/sidebar.module';
