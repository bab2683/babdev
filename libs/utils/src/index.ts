export * from './lib/css-variables.utils';
