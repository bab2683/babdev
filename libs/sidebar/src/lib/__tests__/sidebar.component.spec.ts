import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import * as rxjs from 'rxjs';

import { SidebarComponent } from '../sidebar.component';
import { SidebarStatus } from '../sidebar.enum';
import * as utils from '../sidebar.utils';

describe('SidebarComponent', () => {
  let component: SidebarComponent;
  let fixture: ComponentFixture<SidebarComponent>;
  let emitSpy;
  let getDirectionSpy: any;
  let determineEventSpy: any;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [SidebarComponent],
      imports: [NoopAnimationsModule],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SidebarComponent);
    component = fixture.componentInstance;
    emitSpy = jest.spyOn(component.currentStatus, 'emit');
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit only initial value', () => {
    component.initialState = SidebarStatus.OPEN;

    fixture.detectChanges();
    component.open();
    expect(emitSpy.mock.calls).toMatchSnapshot();
  });

  describe('emit status', () => {
    it('should emit SidebarStatus.OPENING', () => {
      fixture.detectChanges();
      component.open();
      expect(emitSpy).toHaveBeenCalledWith(SidebarStatus.OPENING);
    });

    it('should emit SidebarStatus.CLOSING', () => {
      component.initialState = SidebarStatus.OPEN;
      fixture.detectChanges();
      component.close();

      expect(emitSpy).toHaveBeenCalledWith(SidebarStatus.CLOSING);
    });
  });

  describe('animationComplete', () => {
    beforeEach(() => {
      fixture.detectChanges();
    });

    it('should change the status to OPEN once OPENING is done', () => {
      component.animationComplete({ toState: SidebarStatus.OPENING } as any);

      expect(emitSpy).toHaveBeenCalledWith(SidebarStatus.OPEN);
    });

    it('should change the status to CLOSE once CLOSING is done', () => {
      component.animationComplete({ toState: SidebarStatus.CLOSING } as any);

      expect(emitSpy).toHaveBeenCalledWith(SidebarStatus.CLOSE);
    });
  });

  describe('touch events', () => {
    it('should not attach touch events', () => {
      component.withTouch = false;
      const combineLatestSpy = jest.spyOn(rxjs, 'combineLatest');

      fixture.detectChanges();

      expect(combineLatestSpy).not.toHaveBeenCalled();
    });

    describe('with events', () => {
      let startEvent: any;
      let moveEvent: any;
      let openSpy: any;
      let closeSpy: any;

      beforeEach(() => {
        getDirectionSpy = jest.spyOn(utils, 'getTouchDirection');
        determineEventSpy = jest.spyOn(utils, 'determineEvent');
        openSpy = jest.spyOn(component, 'open');
        closeSpy = jest.spyOn(component, 'close');
        startEvent = { touches: [{ clientX: 0 }] };
        moveEvent = { touches: [{ clientX: 20 }] };
      });

      it('should open the sidebar', () => {
        jest
          .spyOn(rxjs, 'fromEvent')
          .mockReturnValueOnce(rxjs.of(startEvent))
          .mockReturnValueOnce(rxjs.of(moveEvent));

        fixture.detectChanges();

        expect(getDirectionSpy).toHaveBeenCalled();
        expect(determineEventSpy).toHaveBeenCalled();
        expect(openSpy).toHaveBeenCalled();
      });

      it('should close the sidebar', () => {
        jest
          .spyOn(rxjs, 'fromEvent')
          .mockReturnValueOnce(rxjs.of(moveEvent))
          .mockReturnValueOnce(rxjs.of(startEvent));

        fixture.detectChanges();

        expect(getDirectionSpy).toHaveBeenCalled();
        expect(determineEventSpy).toHaveBeenCalled();
        expect(closeSpy).toHaveBeenCalled();
      });
    });
  });
});
