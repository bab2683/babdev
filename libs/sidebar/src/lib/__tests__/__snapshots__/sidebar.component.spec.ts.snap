// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarComponent should emit only initial value 1`] = `
Array [
  Array [
    "open",
  ],
]
`;

exports[`SidebarComponent should match snapshot 1`] = `
<babdev-sidebar
  classPrefix={[Function String]}
  closeButtonIconId={[Function Object]}
  currentStatus={[Function EventEmitter]}
  initialPosition={[Function String]}
  initialState={[Function String]}
  isClose$={[Function Observable]}
  moving="false"
  position={[Function String]}
  sidebarClasses$={[Function Observable]}
  sidebarStaticClasses={[Function Array]}
  sidebarStatus={[Function Object]}
  state$={[Function Observable]}
  stateSubject={[Function BehaviorSubject]}
  status={[Function String]}
  withOverlay={[Function Boolean]}
  withTouch={[Function Boolean]}
>
  <div
    class="babdev-sidebar-overlay ng-tns-c1-0 babdev-sidebar-overlay--hidden ng-trigger ng-trigger-overlayAnimation ng-star-inserted"
    ng-reflect-klass="babdev-sidebar-overlay"
    ng-reflect-ng-class="[object Object]"
    style="opacity:0;0:opacity;"
  /><section
    class="babdev-sidebar ng-trigger ng-trigger-sidebarTranslateAnimation babdev-sidebar--close"
    ng-reflect-klass="babdev-sidebar"
    ng-reflect-ng-class="babdev-sidebar--close"
    style="transform:translateX(-100%);0:transform;"
  >
    <button
      class="babdev-sidebar-close-button"
    >
      <fa-icon
        class="ng-tns-c1-0"
      />
    </button>
  </section>
</babdev-sidebar>
`;
